addons/hw_escpos/escpos/escpos.py

```diff
@@ -886,13 +886,23 @@ def cut(self, mode=''):
 
 
     def cashdraw(self, pin):
-        """ Send pulse to kick the cash drawer """
-        if pin == 2:
-            self._raw(CD_KICK_2)
-        elif pin == 5:
-            self._raw(CD_KICK_5)
-        else:
-            raise CashDrawerError()
+        """ Send pulse to kick the cash drawer
+
+        With some printers the drawer will not open after one puls, for this reason we will check
+        the drawer status up to 5 times and send a new puls if the drawer is still closed. If the
+        drawer status is open, we will not send a pulse.
+        """
+        for i in range(4):
+            printer_status = self.get_printer_status()
+            if not printer_status['printer']['drawer_pin_high']:
```

Comment:
are we sure that the 'deawer_pin_high' is returned by all printers? can't we juste allways send 4 puls?

Reply:
@pimodoo It is returned by all supported printers and drawers, but I'll push a commit with one pulse always send. I guess that the default pulses are very short to spare the solenoid, so sending multiple pulses by default could be a bad idea.